src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,145 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T value) {
+            this.value = value;
+        }
+
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (head == null) {
+            head = tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> newNode = new Node<>(value);
+        Objects.checkIndex(index, size + 1);
+        if (head == null) {
+            head = tail = newNode;
+        } else if (index == 0) {
+            newNode.next = head;
+            head = newNode;
+        } else if (index == size) {
+            tail.next = newNode;
+            tail = newNode;
+        } else {
+            Node<T> previousNode = getNodeByIndex(index - 1);
+            newNode.next = previousNode.next;
+            previousNode.next = newNode;
+        }
```

Comment:
Here you have 4 clauses, it's a bit overengineered, here we need only two clauses - `if (head == null) else {}` . Try to achieve the same effect with less code :)
![image](https://github.com/mate-academy/jv-linked-list/assets/49767049/f0e9df9a-c395-48fc-9e55-142ae5c938b8)